### PR TITLE
ApiManagement - new verison 2020-12-01 update

### PR DESCRIPTION
**Pull request by KacieKK**
## Description

update to latest SDK

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/971
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/968
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/974
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by isra-fel — CHANGES_REQUESTED** [review]
As the log of CI suggests, please fix 2 issues in help documents
```
Target      : New-AzApiManagementKeyVaultObject.md
  Description : Online version in the header of the file is incorrect. The 
                corresponding URL should be the following: https://docs.microsoft
                .com/powershell/module/az.apimanagement/new-azapimanagementkeyvau
                ltobject
  
  Target      : Sync-AzApiManagementKeyVaultSecret.md
  Description : Online version in the header of the file is incorrect. The 
                corresponding URL should be the following: https://docs.microsoft
                .com/powershell/module/az.apimanagement/sync-azapimanagementkeyva
                ultsecret
```

File: src/ApiManagement/ApiManagement/Az.ApiManagement.psd1 [hunk 1]
```diff
@@ -53,7 +53,7 @@ DotNetFrameworkVersion = '4.7.2'
 # ProcessorArchitecture = ''
 
 # Modules that must be imported into the global environment prior to importing this module
-RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '2.4.0'; })
+RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '2.5.1'; })
```
Comment (isra-fel) [inline src/ApiManagement/ApiManagement/Az.ApiManagement.psd1:56]: You needn't worry about this -- it is updated by us whenever there's a new release of your module. So please undo it

File: src/ApiManagement/ApiManagement/Commands/NewAzureApiManagementCustomHostnameConfiguration.cs [hunk 2]
```diff
@@ -56,6 +56,13 @@ public class NewAzureApiManagementCustomHostnameConfiguration : AzureRMCmdlet
             HelpMessage = "KeyVaultId to the secret storing the Custom SSL Certificate.")]
         public string KeyVaultId { get; set; }
 
+        [Parameter(
+            ValueFromPipelineByPropertyName = false,
+            ParameterSetName = SslCertificateFromKeyVault,
+            Mandatory = false,
+            HelpMessage = "User Assigned Managed Identity ClientId credentials to authenticate to KeyVault to fetch Custom SSL Certificate.")]
```
Comment (isra-fel) [inline src/ApiManagement/ApiManagement/Commands/NewAzureApiManagementCustomHostnameConfiguration.cs:63]:
```suggestion
            HelpMessage = "User-assigned managed identity client ID to authenticate to KeyVault to fetch Custom SSL Certificate.")]
```
Client ID is not credential so I suggest remove that word

File: src/ApiManagement/ApiManagement/Commands/AddAzureApiManagementRegion.cs [hunk 3]
```diff
@@ -55,12 +55,32 @@ public class AddAzureApiManagementRegion : AzureApiManagementCmdletBase
...
+            HelpMessage = "A list of availability zones denoting where the api management service is deployed into.")]
+        [ValidateNotNullOrEmpty]
+        public string[] Zone { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Flag only meant to be used for Premium SKU ApiManagement Service and Non Internal VNET deployments. " +
+            "This is useful in case we want to take a gateway region out of rotation." +
+            " This can also be used to standup a new region in Passive mode, test it and then make it Live later.")]
+        public bool? DisableGateway { get; set; }
```
Comment (isra-fel) [inline src/ApiManagement/ApiManagement/Commands/AddAzureApiManagementRegion.cs:70]: We should also mention the default behavior when DIsableGateway is omitted.

File: src/ApiManagement/ApiManagement/Commands/NewAzureApiManagement.cs [hunk 4]
```diff
@@ -133,6 +133,25 @@ public class NewAzureApiManagement : AzureApiManagementCmdletBase
...
+            Mandatory = false,
+            HelpMessage = "Flag only meant to be used for Premium SKU ApiManagement Service and Non Internal VNET deployments. " +
+            "This is useful in case we want to take a gateway region out of rotation." +
+            " This can also be used to standup a new region in Passive mode, test it and then make it Live later.")]
+        public bool? DisableGateway { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Control Plane Apis version constraint for the API Management service.")]
+        public string ApiVersionConstraint { get; set; }
```
Comment (isra-fel) [inline src/ApiManagement/ApiManagement/Commands/NewAzureApiManagement.cs:153]: I believe this string represents the **minimal** API version.
We should name it for example `MinimalApiVersion` or it's too unclear

File: src/ApiManagement/ApiManagement/help/New-AzApiManagementCertificate.md [hunk 5]
```diff
@@ -95,6 +97,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -KeyVault
```
Comment (isra-fel) [inline src/ApiManagement/ApiManagement/help/New-AzApiManagementCertificate.md:100]: Can you add an example that demonstrate the use of `-KeyVault`?

Comment (isra-fel) [reply src/ApiManagement/ApiManagement/help/New-AzApiManagementCertificate.md:100]: Alternatively, add "See `New-AzApiManagementKeyVaultObject` for details." to parameter description

File: src/ApiManagement/ApiManagement/help/New-AzApiManagementNamedValue.md [hunk 6]
```diff
@@ -74,6 +74,21 @@ Accept pipeline input: False
 Accept wildcard characters: False
 ```
 
+### -KeyVault
+KeyVault used to fetch Namedvalue data.This parameter is required if Value not specified.
```
Comment (isra-fel) [inline src/ApiManagement/ApiManagement/help/New-AzApiManagementNamedValue.md:78]: Same comments as in `New-AzApiManagementCertificate.md`

**Comment by isra-fel** [comment]
Besides, some scenario tests are failing

File: src/ApiManagement/ApiManagement/Commands/NewAzureApiManagementCustomHostnameConfiguration.cs (hunk 2 above)
Comment (solankisamir) [reply src/ApiManagement/ApiManagement/Commands/NewAzureApiManagementCustomHostnameConfiguration.cs:63]: #resolve

File: src/ApiManagement/ApiManagement/Commands/AddAzureApiManagementRegion.cs (hunk 3 above)
Comment (solankisamir) [reply src/ApiManagement/ApiManagement/Commands/AddAzureApiManagementRegion.cs:70]: #resolve

**Comment by KacieKK** [comment]
@isra-fel Please take a look again

**Review by solankisamir — APPROVED** [review]

**Comment by solankisamir** [comment]
@isra-fel can you please take a look at this?

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**